pkg/profile/builders/builders.go

```diff
@@ -17,5 +17,5 @@ const (
 //go:generate counterfeiter -o fakes/builder_maker.go . Builder
 type Builder interface {
 	// Build a single artifact from a profile artifact and installation.
-	Build(artifact profilesv1.Artifact, installation profilesv1.ProfileInstallation, definition profilesv1.ProfileDefinition) ([]artifact.Artifact, error)
+	Build(artifact profilesv1.Artifact, installation profilesv1.ProfileInstallation, definition profilesv1.ProfileDefinition, dependencies []profilesv1.Artifact) ([]artifact.Artifact, error)
```

Comment:
@aclevername @Callisto13 so I didn't find a better solution than to extend the interface for easy handling of the dependencies. I'm not sure I like this. But it might be fine. I don't know. :D WDYT?

I would rather have it on the builder, but we aren't constructing the builders at that point. It makes sense to construct the list in the make artifacts too because that's what has all the artifacts. So this might be fine, but if you have a better idea, feel free to suggest. :)

Reply:
do we need an extra argument? I thought perhaps it already had all the information it needed being passed in

Reply:
It's in the definition. WTF am I even doing.

Reply:
Oh Gods, this hurts so much man. I don't even know. I blame it on my meds.